packages/components/src/components/Select/useSelectData.tsx

```diff
@@ -53,42 +77,39 @@ export const useSelectData = (children: ReactNode) => {
                     }
                 }
             });
-
-            return {
-                inputSlots,
-                menuSlots,
-                clearButton,
-                itemValues: getRecursiveOptionValues(menuSlots),
-            };
         } else {
-            return {
-                menuSlots: children,
-                inputSlots: [],
-                itemValues: getRecursiveOptionValues(children),
-            };
+            menuSlots.push(children);
         }
+
+        return {
+            inputSlots,
+            menuSlots,
+            clearButton,
+        };
     }, [children]);
 
-    const filteredItems = useMemo(
-        () =>
-            itemValues.filter(
-                (item) => filterText === '' || item.label.toLowerCase().includes(filterText.toLowerCase()),
-            ),
-        [itemValues, filterText],
-    );
+    const itemValues = useMemo(() => getRecursiveOptionValues(menuSlots), [menuSlots]);
```

Comment:
good change! makes it more readable 👍🏻

Reply:
reverted it though, I did this because when I moved the new logic into this hook I needed this to exist prior to the initialization of the slots but now there is no reason to keep it there as I improved the structure to make this not be necessary